vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java

```diff
@@ -103,14 +104,19 @@ public boolean isInputValid() {
   }
 
   public void setExperiment(Experiment experiment) {
-    SampleRegistrationSpreadsheet.setExperimentMetadata(experiment);
+    this.experiment = experiment;
     experimentName.setText(experiment.getName());
+    SampleRegistrationSpreadsheet.setExperimentMetadata(experiment);
   }
 
   public List<SampleRegistrationContent> getContent() {
     return sampleInformationLayoutHandler.getContent();
   }
 
+  public Experiment getExperiment() {
```

Comment:
Since experiment is an entity not an DTO, this is not side effect free. Why is this needed?

Reply:
Good point I'll expose only the experimentID